### DISCUSSION TITLE
Extend expression parser with support for regular expressions

### DISCUSSION THREAD

**Discussion by sergiubologa** [josdejong/mathjs#2706] [Ideas]
For the code bellow:
`math.eval('typeof(/a regexp/)')`

the library throws the following error:
`Uncaught SyntaxError: Value expected (char 8)`

The following code works just fine:
`math.eval('typeof(123)')` => `'number'`
`math.typeof(/a regexp/)` => `RegExp`

Is this an issue?

**Comment by josdejong** [comment]
The expression parser does not support regular expressions, therefore it throws a syntax error "Value expected (char 8)".

If you need support for regexp you could create a function for it and use that:

``` js
math.import({
  regexp: math.typed({
    'string': function (pattern) {
      return new RegExp(pattern);
    },
    'string, string': function (pattern, flags) {
      return new RegExp(patter, flags);
    }
  })
});

// create a regexp testing whether a string contains only digits
math.eval('r=regexp("^\\\\d+$")', scope);

// use the created regexp
math.eval('r.test("123")', scope);  // returns true
math.eval('r.test("a23")', scope);  // returns false
```

**Comment by sergiubologa** [comment]
Ok, thanks for you answer. 
Is there any plan to add support for regular expressions?

**Comment by josdejong** [comment]
There are no short term plans to integrate support for regular expressions. You're the first one with this request. I will change the title of this issue to "Support for RegExp" and mark it as a feature request.
